.github/workflows/ci.yml

```diff
@@ -32,6 +32,12 @@ jobs:
       - uses: actions/checkout@v4
       - name: Bootc Ubuntu Setup
         uses: ./.github/actions/bootc-ubuntu-setup
+      - name: Setup Rust cache
+        uses: Swatinem/rust-cache@v2
+        with:
+          cache-all-crates: true
+          save-if: ${{ github.ref == 'refs/heads/main' }}
```

Comment:
Why? My understanding is that due to how github handles caching we're safe against e.g. malicious PRs.

Reply:
This will ensure that caches are saved only on `main`, not on any other PRs. I think that's enough caching. It may also have some vague security benefits.
But if you don't like it, it doesn't have to be there.